src/hooks/useOutputMaterial.ts

```diff
@@ -1,34 +1,69 @@
-import { IGeometryData, IMaterialAbstractData, IMaterialStandardDataProperties, IOutputApi, ITreeNode, MaterialStandardData } from "@shapediver/viewer";
-import { useCallback, useEffect, useRef, useState } from "react";
+import { 
+	GeometryData, 
+	IGeometryData, 
+	IMaterialAbstractData, 
+	IMaterialGemDataProperties, 
+	IMaterialSpecularGlossinessDataProperties, 
+	IMaterialStandardDataProperties, 
+	IMaterialUnlitDataProperties, 
+	IOutputApi, 
+	ITreeNode,
+	MaterialGemData, 
+	MaterialSpecularGlossinessData, 
+	MaterialStandardData, 
+	MaterialUnlitData 
+} from "@shapediver/viewer";
+import { useCallback, useEffect, useRef } from "react";
 import { useOutputNode } from "./useOutputNode";
 
+export enum MaterialType {
+	Standard = "Standard",
+	SpecularGlossiness = "SpecularGlossiness",
+	Unlit = "Unlit",
+	Gem = "Gem"
+}
```

Comment:
@MajorMeerkatThe3rd a last comment: It would be nicer if we wouldn't have to use these types, and instead have a simple way to differentiate between material types as part of `IMaterialAbstractDataProperties`. The `createMaterial` function would ideally also be implemented and exported from the viewer. Do you agree? If so, please create a task and link it here.

Reply:
@snabela I understand why you want this, but you would need the types for this either way. As the properties of a material definition can fit to multiple types. 

Example: The properties `{color: '#ff0000'}` are valid material properties for all types, so this could be the properties of a standard material or an unlit material. We don't know.

So basically I could only move the `createMaterial`-function that you currently have in the react example to the viewer side. Then I would move the MaterialTypes enum to the viewer as well. Would you prefer that?

Reply:
@MajorMeerkatThe3rd I would suggest doing sth like this: 

On `IMaterialAbstractDataProperties` add an enum property `type: MaterialType`. 

On the interfaces `IMaterialStandardDataProperties` etc overwrite the `type` by the specific one. 

Essentially we just make the type part of the interface.  I know this is a breaking change because we would now require `type` to be there, but it makes sense I think.